### PR TITLE
Large PR: Make use of smart-pointers for Database::Query

### PR DESCRIPTION
**Pull request by 0blu**
## 🍰 Pullrequest
This pull request will change every occurrence of `QueryResult*` to a smart pointer.

This will ensure that no `QueryResult` might ever accidently leak memory.

### CODE REVIEW

File: src/shared/Database/SqlDelayThread.cpp [hunk 1]
```diff
@@ -68,8 +68,7 @@ void SqlDelayThread::run()
         {
             loopCounter = 0;
             m_dbEngine->Ping();
-            if (QueryResult* res = m_dbConnection->Query("SELECT 1"))
-                delete res;
+            /* ignore result */ m_dbConnection->Query("SELECT 1"); // TODO: Why is this here, when its already check in m_dbEngine->Ping(); ???
```
Comment (0blu) [inline src/shared/Database/SqlDelayThread.cpp:71]: Can someone explain what this check does?

File: src/shared/Database/QueryResult.h [hunk 2]
```diff
@@ -53,8 +53,8 @@ typedef std::vector<std::string> QueryFieldNames;
 class QueryNamedResult
 {
     public:
-        explicit QueryNamedResult(QueryResult* query, QueryFieldNames const& names) : mQuery(query), mFieldNames(names) {}
-        ~QueryNamedResult() { delete mQuery; }
+        explicit QueryNamedResult(std::unique_ptr<QueryResult> query, QueryFieldNames names) : mQuery(std::move(query)), mFieldNames(std::move(names)) {}
+        ~QueryNamedResult() = default;
```
Comment (0blu) [inline src/shared/Database/QueryResult.h]: The destructor declaration can technically be removed.

**Comment by 0blu** [comment]
Where is `WorldSession::SendNameQueryOpcodeFromDBCallBack` used?

**Comment by ratkosrb** [comment]
Seems to have been previously used in SendNameQueryOpcodeFromDB.

**Pull request merged**